### DISCUSSION TITLE
Visidata and Virtual Environments

### DISCUSSION THREAD

**Discussion by ecalifornica** [saulpw/visidata#2427] [Q&A]
Hello! I'm curious whether or not it's possible to use Visidata with a virtualenv? I'm unable to get Visidata to find my Pandas install.

**Comment by ajkerrigan** [comment]
It's possible, sure, but the approach varies a bit depending on how you installed VisiData.

The cleanest way I know of is to install VisiData into its own virtual environment from the start, and install loader dependencies into that same environment. If you already have an analysis environment with Pandas installed, pip installing VisiData there would be another option. Or if you've already installed through something other than pip (like Homebrew or apt), you can sniff your way from the `vd` script back to the Python environment it belongs to. And use that to install packages:

```bash
$(head -1 $(which vd) | tr -d '#!') -m pip list
$(head -1 $(which vd) | tr -d '#!') -m pip install pandas
```

**Reply by ecalifornica** [reply]
Oh that is slick, thank you! I ended up nuking it from orbit with `apt install python3-pandas`, your approach would've been better. Thanks for the confirmation that it was a problem with my setup, super helpful.

**Reply by ajkerrigan** [reply]
> Oh that is slick, thank you! I ended up nuking it from orbit with `apt install python3-pandas`, your approach would've been better. Thanks for the confirmation that it was a problem with my setup, super helpful.

Haha, nothing wrong with nuking from orbit either. :sunglasses:
